### PR TITLE
Reduce the footprint and interface of STObject and STLedgerEntry

### PR DESCRIPTION
**Pull request by nbougalis**

### CODE REVIEW

File: src/ripple/protocol/STObject.h [hunk 1]
```diff
@@ -440,7 +437,6 @@ class STObject
     STPathSet const& getFieldPathSet (SField const& field) const;
```
Comment (willwray) [inline src/ripple/protocol/STObject.h]: Just an observation that `getFieldPathSet` remains while `setFieldPathSet` is removed.

Comment (nbougalis) [inline src/ripple/protocol/STObject.h]: Yep - `getFieldPathSet` is used in a few places, whereas `setFieldPathSet` isn't, so I chose to retain one and ditch the other.

**Comment by vinniefalco** [comment]
"footprint" is borderline non-sterile

**Comment by HowardHinnant** [comment]
👍

File: src/ripple/protocol/STLedgerEntry.h [hunk 2]
```diff
@@ -87,42 +84,19 @@ class STLedgerEntry final
...
-
-    uint256 getThreadedTransaction () const;
-
-    std::uint32_t getThreadedLedger () const;
+    // is this a ledger entry that can be threaded
+    bool isThreadedType() const;
 
-    bool thread (uint256 const& txID, std::uint32_t ledgerSeq, uint256 & prevTxID,
-                 std::uint32_t & prevLedgerID);
+    bool thread (
```
Comment (vinniefalco) [inline src/ripple/protocol/STLedgerEntry.h]: What do you think about `push_front` instead of `thread`? Just an idea

**Comment by vinniefalco** [comment]
Good tidying. Suggested commit message:

```
Refactor STObject, STLedgerEntry:

* Normalize names
* Remove unused and deprecated members
```

**Comment by willwray** [comment]
:+1:

**Comment by nbougalis** [comment]
Updated commit message, rebased. Marked as "Passed"

**Comment by nbougalis** [comment]
Merged as 9a988963e9198d9517984f75c6e48fd98b6cf6f5

**Pull request closed without merging**